### PR TITLE
[bitnami/redis] Add seperate serviceAccunts to master and replicas

### PR DESCRIPTION
**Pull request by adrianNEMO**
### Description of the change

This gives the master and replicas separate service accounts.

### Benefits

 This allows running Redis in a service mesh like consul.

### Possible drawbacks

It would be breaking for people who rely on having the same service account for the master and the replicas.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by rafariossaa** [comment]
Hi,
  Thanks for providing this.
  To not break compatibility, What about using the current `serviceAccountname` for both (master and replicas) when not specific one is set ?.

**Comment by adrianNEMO** [comment]
> Hi, Thanks for providing this. To not break compatibility, What about using the current `serviceAccountname` for both (master and replicas) when not specific one is set ?.

Hi, good Idea I have updated the chart to only use the specific service account when it is manually enabled in the master or replica config.

File: bitnami/redis/Chart.yaml [hunk 1]
```diff
@@ -24,4 +24,4 @@ maintainers:
 name: redis
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/redis
-version: 17.0.10
+version: 18.0.0
```
Comment (rafariossaa) [inline bitnami/redis/Chart.yaml:27]: In this case, a major version is not longer needed.

**Review by rafariossaa — CHANGES_REQUESTED** [review]

**Comment by rafariossaa** [comment]
Also, you will need to rebase you changes from master.

File: bitnami/redis/Chart.yaml (hunk 1 above)
Comment (adrianNEMO) [reply bitnami/redis/Chart.yaml:27]: :heavy_check_mark:

**Comment by adrianNEMO** [comment]
> Also, you will need to rebase you changes from master.

:heavy_check_mark:

**Review by rafariossaa — APPROVED** [review]
Thanks for providing this.
LGTM

**Pull request merged**